java/src/jmri/jmrix/dccpp/DCCppTrafficController.java

```diff
@@ -22,15 +22,24 @@
  * Based on XNetTrafficController by Bob Jacobsen and Paul Bender
  */
 public abstract class DCCppTrafficController extends AbstractMRTrafficController implements DCCppInterface {
-
+    
+    public int startUpDelay = 1500; //in ms, will be overridden by config option
+    public boolean anyReceived = false;  //will be turned on as soon as any incoming traffic seen
+    
     @Override
     protected void transmitLoop() {
-        log.debug("Don't start sending for 1.5 seconds to avoid Arduino restart");
-        try {
-            Thread.sleep(1500);
-        } catch (InterruptedException ignore) {
-            Thread.currentThread().interrupt();
+        int totalDelay = 0;
+        int loopDelay = 100;
+        log.debug("Don't start sending for up to {}ms to avoid Arduino restart", startUpDelay);
+        while (!anyReceived && (totalDelay <= startUpDelay)) {
+            try {
+                Thread.sleep(loopDelay);
+            } catch (InterruptedException ignore) {
+                Thread.currentThread().interrupt();
+            }
+            totalDelay += loopDelay;
         }
```

Comment:
@bobjacobsen 
What happens if an InterruptedException is thrown while in the sleep() method? I'm well aware that the JMRI [recommended practices](https://www.jmri.org/help/en/html/doc/Technical/Threads.shtml) tells to never swallow an InterruptedException.

But my concern in this case is that it happens in a loop. My concern is that if it happens, the catch statement will catch it, rethrow it, and then it will throw again in the sleep method, be catched and rethrown and then thrown again in the sleep method. Causing an endless loop with 100% cpu.

But I might be totally wrong here so I want to ask.

Reply:
That's a good question.  In this case, the loop will zip around adding `loopDelay` each (quick) cycle until the `while` condition is satisfied.  You could bypass that by adding a check of the interrupted flag to the `while` condition if you wanted to explicitly handle that.

Reply:
Or, even simpler, you could add a `break` in the `catch` clause.

Reply:
Thanks for pointing this out, I've little experience in this area. 
I will add `break` after `Thread.currentThread().interrupt();` Is that correct?
I admit I have no idea what that `interrupt()` does, nor do I know how to force the exception to test it.

Reply:
> I will add `break` after `Thread.currentThread().interrupt();` Is that correct?

Yes. I suggest that you add a comment that the break breaks the while loop. It shouldn't be needed but it helps us who are a bit unsure about interrupted threads.

`break; // Breaks the while loop`

> I admit I have no idea what that `interrupt()` does, nor do I know how to force the exception to test it.

Things in Java that might block, for example Thread.sleep() or reading from a stream, can be aborted premature by calling the method [interrupt()](https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#interrupt--) on its thread.

When you call `Thread.currentThread().interrupt()` when the thread is not blocked, like in this case, you tell Java to interrupt the thread the next time Java does something that blocks the thread, for example Thread.sleep() or waiting on a stream to read data.

Reply:
Done, thanks! PR updated.